x-pack/platform/plugins/shared/ml/public/alerting/anomaly_detection_rule/ml_anomaly_alert_trigger.tsx

```diff
@@ -137,6 +147,30 @@ const MlAnomalyAlertTrigger: FC<MlAnomalyAlertTriggerProps> = ({
     [jobsAndGroupIds]
   );
 
+  useEffect(
+    function handleFilterPreservationAcrossResultTypes() {
```

Comment:
why do we need this effect? it is quite difficult to follow. is it here to restore the prev filter value when the user is clicking on different result type tiles?